app/models/repository.rb

```diff
@@ -61,12 +61,14 @@ def self.add_repo(event, namespace, repo, tag)
     if repository.nil?
       repository = Repository.create(namespace: namespace, name: repo)
     elsif repository.tags.exists?(name: tag)
-      return
+      repository.tags.where(name: tag).each { |tag| tag.touch }
```

Comment:
This would be better:
```ruby
return unless repository.tags.find_by(name: "latest").try(:touch)
```

This takes into account:

1. The method returns nil on failure.
2. The name of a tag inside of a repo is unique.